### DISCUSSION TITLE
Error in gh script pushing geometry to ETABS 21

### DISCUSSION THREAD

**Discussion by jamesangevine** [BHoM/BHoM#1612] [General]
Hi All! Apologies in advance if this is the wrong location to bring up this question. 

I've been trying to get up to speed on using BHoM for ETABS model creation, and I've been running into a consistent issue on a number of example scripts, each one getting more and more simple. 

![image](https://github.com/BHoM/BHoM/assets/109187754/c1511956-7023-467b-9fbc-de4d27ff5bd5)

The error always pops up when I try to push the geometry to ETABs, with the fatal error as follows: 

1. This component failed to run properly.
- Error: Unable to cast object of type 'System.Boolean' to type 'System.Double[]'.
- Occured in  BH.UI.Base.Caller.Run(List`1 inputs)
     called from BH.UI.Base.Caller.Run()
- Are you sure you have the correct type of inputs? Check their description for more details.

I've included the gh file I created below, but it's essentially the same gh script shown on page 19 of this document: https://static.au-uw2-prd.autodesk.com/Class_Handout_BES500065_ClassHandout-BES500065-Rocha-AU2021-V2.pdf

https://drive.google.com/file/d/1CTx8rifpiq6uMJPUQpH6uxiV_RXuw4aS/view?usp=sharing

I'm currently using Rhino 7 SR36, Grasshopper Build 1.0.0007, BHoM 7.1.1.0, ETABS Ultimate 64 Bit Version 21.0.1 Build 3178 

I recognize ETABS version 21 isn't supported currently by the toolkit, but I'm hoping (fingers crossed), that it's not an ETABS issue.

Please let me know if there is any additional information I need to provide, thanks!

**Comment by chandrasajana** [comment]
on my PC, your codes are working fine and I'm able to push the model into ETABS v21 with just a section warning (can be ignored)

![Screenshot 2024-04-28 225210](https://github.com/BHoM/BHoM/assets/93826774/81284bad-ec61-488f-85b5-fb603b8c89c6)

**Comment by samuelbasimalla** [comment]
Hi @jamesangevine 
Though it has not been tested, the ETABS adapter should most likely work for Version 21(as shown above by @chandrasajana).
You can see from ETABS API release notes (screencap below), CSI has done away with version number in its dll (beginning from its Version 18) and ETABS toolkit should automatically reference the latest version installed on machine.
![image](https://github.com/BHoM/BHoM/assets/148187206/da9e5040-07d2-4007-ad2c-28a28f55912d)

**Comment by udhay111** [comment]
Just open blank stabs windows with blank template and run the boolean i too have this error and checked your file running on bhom etabs 22.7
